### PR TITLE
CLOUDP-67147: mongocli atlas cluster(s) search index(es) list|ls

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [CLOUDP-67147](https://jira.mongodb.org/browse/CLOUDP-67147)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: e2e/atlas/search_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			atlasEntity,
+			clustersEntity,
+			searchEntity,
+			"list",
+			"--clusterName="+clusterName,
+			"--db=test",
+			"--collection=test")
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
```
Comment (robcarlan-mlab) [inline e2e/atlas/search_test.go:57]: Can we do a check on the response here?

File: internal/store/search.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+//go:generate mockgen -destination=../mocks/mock_search.go -package=mocks github.com/mongodb/mongocli/internal/store SearchIndexLister
+
+type SearchIndexLister interface {
+	SearchIndexes(string, string, string, string, *atlas.ListOptions) ([]*atlas.SearchIndex, error)
+}
+
+// OnlineArchives encapsulate the logic to manage different cloud providers
```
Comment (robcarlan-mlab) [inline internal/store/search.go:31]: copy paste comment

File: e2e/atlas/search_test.go (hunk 1 above)
Comment (gssbzn) [reply e2e/atlas/search_test.go:57]: it would be an empty array so I don't know how helpful that'll be

Comment (robcarlan-mlab) [reply e2e/atlas/search_test.go:57]: ah ok

**Pull request merged**